File: src/main/scala/Day1.scala
object Day1 {
  val input = scala.io.Source.fromFile("data/data1.txt", "UTF-8").getLines.toVector

  def part1(): Unit = println("Answer:" + input.map(_.toInt/3 - 2).sum)

  def part2(): Unit = {
    var sum = 0
    def rec(m: Int): Unit = if (m > 0) {sum += m; rec(m/3 - 2)}

    input.foreach(m => rec(m.toInt/3 - 2))
    println("Answer:" + sum)
  }

  def apply() = {
    println("SOLUTION DAY 1")
    println("Running part 1")
    part1()
    println("Running part 2")
    part2()
  }
}

File: 2020/src/aoc2020/Day1.java
package aoc2020;

import java.io.*;
import java.util.*;

public class Day1 {
	private List<Integer> input = new ArrayList<>();
	
	Day1() throws FileNotFoundException {
		Scanner scan = new Scanner(new File("data/data1.txt"));
		while (scan.hasNextLine()) {
			input.add(Integer.parseInt(scan.nextLine()));
		}
	}
	
	private void part1() {
		for (int i = 0; i < input.size(); i++) {
			for (int j = i + 1; j < input.size(); j++) {
				if (input.get(i) + input.get(j) == 2020) {
					System.out.println(input.get(i) * input.get(j));
					return;
				}
			}
		}
	}
	
	private void part2() {
		for (int i = 0; i < input.size(); i++) {
			for (int j = i + 1; j < input.size(); j++) {
				for (int k = j + 1; k < input.size(); k++) {
					if (input.get(i) + input.get(j) + input.get(k) == 2020) {
						System.out.println(input.get(i) * input.get(j) * input.get(k));
						return;
					}
				}
			}
		}
	}
	
	public static void main(String[] args) throws Exception {
		Day1 day = new Day1();
		day.part1();
		day.part2();
	}

}